src/plugins/usage_collection/public/plugin.tsx

```diff
@@ -150,7 +150,8 @@ export class UsageCollectionPlugin
     if (
       this.config.uiCounters.enabled &&
       !isUnauthenticated(http) &&
-      !screenshotMode.isScreenshotMode()
+      !screenshotMode.isScreenshotMode() &&
+      !window.navigator.userAgent.includes('Elastic/Synthetics')
```

Comment:
NIT: I feel like this (`window.navigator.userAgent.includes('Elastic/Synthetics')`) should be extracted and factorized to something like `function isSyntheticsAgent()`. (or at least the UA string?)

Is there a package this could be exposed from?

Reply:
I couldn't find any suitable package, so I created a new package `@kbn/analytics-collection-utils` in https://github.com/elastic/kibana/pull/171054/commits/f5ba773f8b65c00c27fd615e6e40bd9d09f97fa8